zoom_map.py

```diff
@@ -79,26 +88,57 @@ def add_level(level):
                 [("<Button-4>", partial(self._map_zoom, -1)),
                  ("<Button-5>", partial(self._map_zoom,  1)),
                  ("<Button-1>", self._on_click),
-                 ("<B1-Motion>", self._on_drag)]]
+                 ("<B1-Motion>", self._on_drag),
+                 ("<ButtonRelease-1>", self._on_unclick)]]
 
     def _set_image(self):
-        self._tk_image = self.create_image(0, 0, anchor=tk.NW,
-                                           image=self.image)
+        if self._tk_image is not None:
+            # Clean up the old image.
+            self.delete(self._tk_image)
```

Comment:
I agree that use of the `del` operator is frowned upon, but this is a call to `tkinter.Canvas.delete`, which removes the image from the Canvas' internal stack. [SO](http://stackoverflow.com/questions/15839491/how-to-clear-tkinter-canvas#15840231) and [the tkinter docs](http://effbot.org/tkinterbook/canvas.htm) both seem to advocate using `.delete()`. However, I'm new to tkinter and might have misinterpreted. You've got more experience with GUIs than I do; is it unusual to explicitly delete elements like this?

I tried removing the call to delete, and things still seem to work and I don't obviously have a memory leak (but maybe that's because the `PhotoImage` object doesn't store its own reference to the PIL image, and maybe it cleans itself up when the `PhotoImage` is no longer available, or something). So, it does seem safe to remove if that's typical.